database/db.go

```diff
@@ -769,6 +769,30 @@ func (db *DB) Delete(
 	return db.DeleteStreamed(ctx, entityType, idsCh, onSuccess...)
 }
 
+// ExecTx allows running a function in a database transaction without requiring manual transaction handling.
```

Comment:
Description would benefit from simple english and simple tenses, e.g. "allows running a function" vs. "executes", "is started" vs "starts".

I would write the following:

ExecTx executes a function within a database transaction.

Starts a new transaction, executes the provided function, and commits the transaction if the function succeeds. If the function returns an error, the transaction is rolled back.

Returns an error if starting the transaction, executing the function, or committing the transaction fails.